config/config.go

```diff
@@ -25,195 +25,230 @@ import (
-	CA string `yaml:"ca"`
+	// EnableProfiling represents whether to enable profiling functionality.
+	Profiling bool `yaml:"profiling"`
 }
 
-// Athenz represents the configuration for authorization proxy server to connect to Athenz.
+// Athenz represents the Athenz server connection configuration.
 type Athenz struct {
-	// URL represents the Athenz (ZMS and ZTS) URL handle authentication and authorization request.
+	// URL represents the Athenz (ZMS or ZTS) API URL.
 	URL string `yaml:"url"`
 
 	// Timeout represents the request timeout duration to Athenz server.
 	Timeout string `yaml:"timeout"`
 
-	// AthenzRootCA is the environment variable name having the Athenz root CA certificate file path for connecting to Athenz.
-	AthenzRootCA string `yaml:"root_ca"`
+	// CAPath represents the CA certificate chain file path for verifying Athenz server certificate.
+	CAPath string `yaml:"caPath"`
 }
 
-// Proxy represent the proxy destination of the authorization proxy.
+// Proxy represents the proxy destination configuration.
 type Proxy struct {
-	// Scheme represent the HTTP URL scheme of the proxy destination, default is http.
+	// Scheme represents the HTTP URL scheme of the proxy destination, default is http.
 	Scheme string `yaml:"scheme"`
 
-	// Host represent the proxy destination host, for example localhost.
+	// Host represents the proxy destination host, for example, localhost.
 	Host string `yaml:"host"`
 
-	// Port represent the proxy destination port number.
+	// Port represents the proxy destination port number.
 	Port uint16 `yaml:"port"`
 
-	// RoleHeader represent the HTTP header key name of the role token for Role token proxy request.
-	RoleHeader string `yaml:"role_header_key"`
-
-	// BufferSize represent the reverse proxy buffer size.
-	BufferSize uint64 `yaml:"buffer_size"`
+	// BufferSize represents the reverse proxy buffer size.
+	BufferSize uint64 `yaml:"bufferSize"`
 
-	// BypassURLPaths represent URL paths that require authorization bypassing.
-	// WARNING!!! Setting this configuration may introduce security hole in your system. Do NOT set this configuration with URL paths other than application's healthcheck endpoint.
-	// Tips for performance: define your healthcheck endpoint with different lenght from the most frequestly used endpoint, e.g. `/most_used` (len: 10) with `/healthcheck` (len: 12), instead of `/healthccc` (len: 10)
-	BypassURLPaths []string `yaml:"bypass_url_paths"`
+	// BypassURLPaths represents URL paths that require to bypass authorization.
+	// WARNING!!! Setting this configuration may introduce security hole in your system. ONLY set this configuration as the application's health check endpoint.
+	// Tips for performance: define your health check endpoint with a different length from the most frequestly used endpoint, for example, use `/healthcheck` (len: 12) when `/most_used` (len: 10), instead of `/healthccc` (len: 10)
+	BypassURLPaths []string `yaml:"bypassUrlPaths"`
```

Comment:
@WindzCUHK Since this functionality is not intended to be used for bypassing the authn/authz, `HealchCheckPaths` is more preferred.

Reply:
Seems it is under `proxy:`, should it be `AppHealthCheckPaths`/`OriginHealthCheckPaths`?
Or, move the config under `authorization:`?

Reply:
@tatyano

Reply:
@tatyano Any idea on the new name?

Reply:
@WindzCUHK I think `OriginHealthCheckPaths` looks good. `AppHealthCheckPaths` does not really tell what is `App`. Moving it under `authorization:` is optional for me.

Reply:
fixed in: 25c83be